### PR TITLE
Use arbitrary length integers to timestamp snapshots

### PR DESCRIPTION
**Pull request by pbartholomew08**
This will give snapshot files of the form

ux1
ux2
ux3
...
uxN

without zero padding, currently checkpoints/statistics have not been considered.

### CODE REVIEW

File: src/visu.f90 [hunk 1]
```diff
@@ -320,7 +322,8 @@ subroutine end_snapshot(itime, num)
         write(fmt3,'("(A,F16.4)")')
         write(fmt4,'("(A,F16.12)")')
 
-        open(newunit=is,file="./data/snap"//trim(num)//".ini",action='write',status='replace')
+        write(filename, '(A, I0, A)') "./data/snap", num, ".ini"
```
Comment (mathrack) [inline src/visu.f90]: This statement allocates the character array "filename" ?

File: src/visu.f90 [hunk 2]
```diff
@@ -576,22 +581,23 @@ subroutine write_field(f1, pathname, filename, num, skip_ibm, flush)
...
   function gen_filename(pathname, varname, num, ext)
 
-    character(len=*), intent(in) :: pathname, varname, num, ext
+    character(len=*), intent(in) :: pathname, varname, ext
+    integer, intent(in) :: num
 #ifndef ADIOS2
-    character(len=(len(pathname) + 1 + len(varname) + 1 + len(num) + 1 + len(ext))) :: gen_filename
-    write(gen_filename, "(A)") pathname//'/'//varname//'-'//num//'.'//ext
+    character(len=:), allocatable :: gen_filename
+    gen_filename = pathname//'/'//varname//'-'//int_to_str(num)//'.'//ext
```
Comment (mathrack) [inline src/visu.f90:577]: This statement allocates the character array "gen_filename" ?

File: src/visu.f90 [hunk 3]
```diff
@@ -601,17 +607,25 @@ end function gen_filename
...
 #ifndef ADIOS2
     character(len=*), parameter :: path_to_h5file = "./"
     character(len=(len(path_to_h5file) + len(filename))) :: gen_h5path
     write(gen_h5path, "(A)") path_to_h5file//filename
 #else
     character(len=*), parameter :: path_to_h5file = "../data.hdf5:/Step"
-    character(len=(len(path_to_h5file) + len(num) + 1+ len(filename))) :: gen_h5path
-    write(gen_h5path, "(A)") path_to_h5file//num//"/"//filename
+    character(len=:), allocatable :: gen_h5path
+    gen_h5path = path_to_h5file//int_to_str(num)//"/"//filename
```
Comment (mathrack) [inline src/visu.f90:596]: This statement allocates the character array "gen_h5path" ?

File: src/visu.f90 (hunk 2 above)
Comment (pbartholomew08) [reply src/visu.f90:577]: Yes - allocatable intrinsic types are (re)allocated on assignment (F2003).

File: src/visu.f90 (hunk 3 above)
Comment (pbartholomew08) [reply src/visu.f90:596]: See https://github.com/xcompact3d/Incompact3d/pull/94#discussion_r795452179

File: src/visu.f90 (hunk 1 above)
Comment (pbartholomew08) [reply src/visu.f90]: I need to double check this one - it didn't cause any issues though

File: src/visu.f90 (hunk 2 above)
Comment (mathrack) [reply src/visu.f90:577]: Nice

**Comment by pbartholomew08** [comment]
Replaces ``write(str, fmt) ...`` with ``str = ...`` where ``str`` is an allocatable character array - it isn't clear that the former is either correct or portable. (See https://github.com/xcompact3d/Incompact3d/pull/94#discussion_r795452667)

**Comment by slaizet** [comment]
Paul, is this formatting compatible with Paraview?

**Comment by pbartholomew08** [comment]
Some code in LES models had drifted out of consistency, merged current master into branch and tested with ParaView (20 snapshots covering single and double digits)

**Pull request merged**